contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,441 @@
+        uint256[] storage depositIds = depositorIds[donor_];
+        if (depositIds.length == 0) {
+            return 0;
+        }
+
+        for (uint256 index = 0; index < depositIds.length; index++) {
+            uint256 id = depositIds[index];
+
+            if (recipientLookup[id] == recipient_) {
+                return IgOHM(gOHM).balanceTo(depositInfo[id].principalAmount);
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+        @notice Return total amount of donor's gOHM deposited (updated to current index based on sOHM equivalent amount deposits)
+        @param donor_ Address of user donating yield
+     */
+    function totalDeposits(address donor_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+        uint256 principalTotal = 0;
+
+        for (uint256 index = 0; index < depositIds.length; index++) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+            principalTotal += currDeposit.principalAmount;
+        }
+
+        return IgOHM(gOHM).balanceTo(principalTotal);
+    }
+
+    /**
+        @notice Return arrays of donor's recipients and deposit amounts (gOHM value based on sOHM equivalent deposit), matched by index
+        @param donor_ Address of user donating yield
+     */
+    function getAllDeposits(address donor_) external view returns (address[] memory, uint256[] memory) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        uint256 len = depositIds.length == 0 ? 1 : depositIds.length;
+
+        address[] memory addresses = new address[](len);
+        uint256[] memory agnosticDeposits = new uint256[](len);
+
+        if (depositIds.length == 0) {
+            addresses[0] = address(0);
+            agnosticDeposits[0] = 0;
+        } else {
+            for (uint256 index = 0; index < len; index++) {
+                DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+                addresses[index] = recipientLookup[depositIds[index]];
+                agnosticDeposits[index] = IgOHM(gOHM).balanceTo(currDeposit.principalAmount);
+            }
+        }
+
+        return (addresses, agnosticDeposits);
+    }
+
+    /**
+        @notice Return total amount of gOHM donated to recipient since last full withdrawal or redemption
+        @param donor_ Address of user donating yield
+        @param recipient_ Address of user recieiving donated yield
+     */
+    function donatedTo(address donor_, address recipient_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        for (uint256 index = 0; index < depositIds.length; index++) {
+            if (recipientLookup[depositIds[index]] == recipient_) {
+                return _getOutstandingYield(depositInfo[depositIds[index]].principalAmount, depositInfo[depositIds[index]].agnosticAmount);
```

Comment:
wouldn't it be simpler to use this function you made
redeemableBalance(uint256 id_) and just chuck in the id. 
makes code look shorter and less chance of error putting in wrong value into _getOutstandingYield()

Reply:
yep, probably

Reply:
done